### ISSUE TITLE
indentation of multi-line #;(commented sexps)

### ISSUE THREAD

**Issue by zenspider** [greghendershott/racket-mode#136]
```racket
#lang racket

;; using:
;;
;;   (defun paredit-semicolon-find-line-break-point () '())
;;
;; removes a lot of my annoyances with paredit's treatment of #;'(...).

'(blah blah
       blah)

;; but indentation is wrong:

#;'(blah blah
blah)

;; especially obvious when using aggressive-indent-mode.
```

extra infos:

```
((emacs-version "24.4.1")
 (system-type darwin)
 (major-mode racket-mode)
 (racket--source-dir "/Users/ryan/.emacs.d/elpa/racket-mode-20150519.737/")
 (racket-racket-program "racket")
 (racket-raco-program "raco")
 (racket-memory-limit 2048)
 (racket-error-context medium)
 (racket-history-filter-regexp "\\`\\s *\\S ?\\S ?\\s *\\'")
 (racket-images-inline t)
 (racket-images-keep-last 100)
 (racket-images-system-viewer "display")
 (racket-pretty-print t)
 (racket-indent-curly-as-sequence t)
 (racket-indent-sequence-depth 0)
 (racket-pretty-lambda nil)
 (racket-smart-open-bracket-enable nil)
 (racket-use-company-mode t)
 (ert--current-run-stats nil)
 (edebug-mode nil)
 (mml-mode nil)
 (aggressive-indent-mode t)
 (racket-check-syntax-mode nil)
 (hs-minor-mode t)
 (erc-track-minor-mode t)
 (elisp-slime-nav-mode nil)
 (paredit-mode t)
 (global-whitespace-newline-mode nil)
 (global-whitespace-mode t)
 (whitespace-newline-mode nil)
 (whitespace-mode nil)
 (global-auto-revert-mode t)
 (auto-revert-tail-mode nil)
 (auto-revert-mode nil)
 (window-number-mode nil)
 (p4-mode nil)
 (diff-minor-mode nil)
 (compilation-minor-mode nil)
 (compilation-shell-minor-mode nil)
 (compilation-in-progress nil)
 (mc-hide-unmatched-lines-mode nil)
 (rectangular-region-mode nil)
 (multiple-cursors-mode nil)
 (visible-mode nil)
 (visual-line-mode nil)
 (next-error-follow-minor-mode nil)
 (abbrev-mode nil)
 (overwrite-mode nil)
 (auto-fill-function nil)
 (defining-kbd-macro nil)
 (isearch-mode nil))
```

**Comment by greghendershott** [comment]
Thanks for the report. I see what you mean. Sexpr comments are tricky, so I'm going to be cautious saying how quickly I can fix this (particularly without creating some new bug), but I'll definitely look at it.

**Comment by greghendershott** [comment]
Sorry this has been open so long.

With the passing of time, and taking a fresh look: Here's my new, hopefully valid insight.

**Sexpr comments should _look_ like comments. But _in every other respect_, they should _not_ act/edit like comments.**

Aside from them appearing in `font-lock-comment-face`, you want everything else in Emacs to work as if they're not comments -- including indentation and navigation within the commented-out sexpr.

In Emacs terms, this means that the "syntax table" for the buffer should _not_ mark them as comments. The `#;` bit should be marked as "prefix" syntax, like `'`. The following text of the sexpr itself should get no special treatment -- it should get the syntax it usually would.

This sounds really obvious now that I say it. Hopefully that means it's correct.

---

This means that I should rip out some (complicated-ish) code I have for handing sexpr comments in the syntax table. And replace it with (what turns out to be simpler) code to do font-lock, only.

TL;DR the correct answer is simpler.

I have some code doing this, now. Although "it works", I'm going to think about and dogfood it awhile before committing and pushing.

**Comment by greghendershott** [comment]
Hmm, I suppose the same could be said about `#| multi-line comments |#`.  Although they're not used very much in idiomatic Racket I've seen, maybe it would be worthwhile trying to handle them similarly.

**Comment by greghendershott** [comment]
Actually. no. I'm going to punt on handling `#| multi-line comments |#` like this. Font-lock doesn't automatically adjust the starting position to include an opening `#|`, the way it does with sexpr comments (via `beginning-of-defun`, IIUC). Although I _could_ make it search back for a possible `#|`, I don't want to make font-lock do more work and be slower for the sake of multi-line comments. I'm not convinced it's worthwhile. At any rate, I think it should be a distinct other issue.

To be clear I do still want to resolve this issue, per se, re sexpr comments. I'm just negating my previous comment.

**Comment by greghendershott** [comment]
I just pushed a commit for this. There was more work than I expected, but, some of it was of general benefit. Every time I go down the rabbit-hole of multi-line font-lock in Emacs, I learn something new I was doing wrong (even if things seemed to work).
